bin/manage.py

```diff
@@ -0,0 +1,406 @@
+    except Exception as e:
+        log.error(e, 'unable to get primary status')
+        sys.exit(1)
+
+    if is_mongo_primary:
+        mongo_update_replset_config(local_mongo, hostname)
+
+# ---------------------------------------------------------
+
+def is_mongo_up(local_mongo, timeout=MONGO_RETRY_TIMES):
+    '''
+    check to see if mongo is up yet, retying the given number of times
+    '''
+    
+    while True:
+        timeout -= 1
+        if timeout <= 0:
+            log.errors('unable to connect to mongodb after %i times' % MONGO_RETRY_TIMES)
+            return False
+
+        try:
+            # check that mongo is up
+            server_info = local_mongo.server_info()
+            if not server_info['ok']:
+                log.info('Mongo response not "ok" on %s; retrying...' % ip)
+                sleep(1)
+                continue
+            break
+        except (AutoReconnect, ServerSelectionTimeoutError) as e:
+            log.info('Mongo not yet available on %s; retrying...' % ip)
+            sleep(1)
+            continue
+        except (ConnectionFailure, NetworkTimeout, NotMasterError) as e:
+            # TODO retry like AutoReconnect error above?
+            log.info(e)
+            return False
+        except Exception as e:
+            # just bail on unexpected exceptions when trying to connect
+            log.error(e)
+            return False
+    return True
+
+@debug
+def mongo_update_replset_config(local_mongo, hostname):
+    '''
+    called from the primary node to update the replica config in mongo
+    using the current set of healthy mongo containers listed in consul
+    '''
+    try:
+        repl_config = local_mongo.admin.command('replSetGetConfig')
+        if not repl_config['ok']:
+            raise Exception('could not get replica config: %s' % repl_config['errmsg'])
+        repl_config = repl_config['config']
+        # TODO use consul.agent.health() instead?
+        consul_services = consul.agent.services()
+
+        mongos_in_consul = []
+        for x in consul_services:
+            host = consul_to_mongo_hostname(consul_services[x]['ID'])
+            if host:
+                mongos_in_consul.append(host)
+        if not len(mongos_in_consul):
+            return
+        if not hostname + ':27017' in mongos_in_consul:
+            return
+
+        max_id = 0
+        changed = False
+        new_members = []
+        for member in repl_config['members']:
+            if member['_id'] > max_id:
+                max_id = member['_id']
+            if member['host'] in mongos_in_consul:
+                mongos_in_consul.remove(member['host'])
+                new_members.append(member)
+            else:
+                changed = True
+        for host in mongos_in_consul:
+            changed = True
+            max_id += 1
+            new_members.append({
+                    '_id': max_id,
+                    'host': host,
+                })
```

Comment:
Yah, the "Pythonic" way is a good deal more compact and generally will use list comprehensions and set operators.

```python
consul_services = consul.agent.services()
mongos_in_consul = [svc['ID'] for svc in consul_services]
if not mongos_in_consul:
    return # empty list from consul means we have nothing to compare against
if not hostname + ':27017' in mongos_in_consul:
    # if the master node is not in the consul services list we need to
    # wait a little longer before configuring mongo
    return

members = repl_config['members']
existing_hosts, ids = zip(*[(member['host'], member['_id']) for member in members])
existing = set(existing_hosts)
current = set(mongos_in_consul)

new_mongos = current - existing
stale_mongos = existing - current

if not new_mongos or not stale_mongos:    
    return # no change

for member in members:
    if member['host'] in stale_mongos:
        members.remove(member)
for new_mongo in new_mongos:
    new_id = max(ids) + 1
    ids.append(new_id)
    members.append({'_id': new_id, 'host': new_mongo})

repl_config['members'] = members
local_mongo.admin.command('replSetReconfig', repl_config)
```

Also, it's not clear to me what the intended behavior is in terms of _id assignment if we add a replica to replace an old one. Disclaimer: I'm not a Mongo expert. But the docs say the _id needs to be unique. If we lose a node and then reassign its _id to a new node, what happens?